src/m365/aad/commands/m365group/m365group-remove.ts

```diff
@@ -115,6 +109,124 @@ class AadM365GroupRemoveCommand extends GraphCommand {
       }
     }
   }
+
+  private async getM365GroupSiteURL(logger: Logger, id: string): Promise<string> {
+    if (this.verbose) {
+      await logger.logToStderr(`Getting the site URL of Microsoft 365 Group: ${id}...`);
+    }
+
+    const requestOptions: CliRequestOptions = {
+      url: `${this.resource}/v1.0/groups/${id}/drive?$select=webUrl`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    const res = await request.get<{ webUrl: string }>(requestOptions);
+    return res.webUrl.substring(0, res.webUrl.lastIndexOf('/'));
+  }
+
+  private async deleteM365GroupSite(logger: Logger, url: string): Promise<void> {
+    if (this.verbose) {
+      await logger.logToStderr(`Deleting the group site: '${url}'...`);
+    }
+
+    const spoAdminUrl = await spo.getSpoAdminUrl(logger, this.debug);
+    this.spoAdminUrl = spoAdminUrl;
+
+    const requestOptions: CliRequestOptions = {
+      url: `${this.spoAdminUrl}/_api/GroupSiteManager/Delete?siteUrl='${url}'`,
+      headers: {
+        'content-type': 'application/json;odata=nometadata',
+        accept: 'application/json;odata=nometadata'
+      },
+      responseType: 'json'
+    };
+
+    await request.post(requestOptions);
+  }
+
+  private async deleteM365GroupFromRecyclebin(logger: Logger, id: string): Promise<void> {
+    const maxRetries = 10;
+    const intervalInMs: number = 6000;
+
+    for (let retries = 0; retries < maxRetries; retries++) {
+      if (await this.isM365GroupInDeletedItemsList(id)) {
+        await this.removeM365GroupPermanently(logger, id);
+        return;
+      }
+      else {
+        if (this.verbose) {
+          await logger.logToStderr(`Group has not been deleted yet. Waiting and retrying...`);
+        }
+
+        await this.sleep(logger, intervalInMs);
+      }
+    }
+
+    await logger.logToStderr(`Group could not be removed from the recycle bin after all retries.`);
```

Comment:
This message is confusing. If the deleting process is still in pending due to extra server load, it could give the impression that we haven't even started deleting, while in fact it's in progress. Let's ensure that we make it clear that the process is pending but haven't finished before the assumed number of retries